spec/models/ems_cluster_spec.rb

```diff
@@ -213,4 +213,36 @@
       expect(result).to be_falsey
     end
   end
+
+  context "tenant_identity" do
+    let(:tenant)   { FactoryGirl.create(:tenant) }
+    # admin user is needed to process Events
+    let(:admin)    { FactoryGirl.create(:user_with_group, :userid => "admin") }
+    let(:ems)      { FactoryGirl.create(:ext_management_system, :tenant => tenant) }
+
+    it "has tenant from provider" do
+      cluster = FactoryGirl.create(:ems_cluster, :ems_id => ems.id)
+      args = {
+        :user_id      => admin.id,
+        :miq_group_id => ems.tenant.default_miq_group.id,
+        :tenant_id    => ems.tenant.id
+      }
+      expect(MiqAeEngine).to receive(:deliver_queue).with(hash_including(args), anything)
```

Comment:
@lfu You do not need to test through MiqAeEngine on each model.  I would expect much simpler tests that call `tenant_identify` with the two configurations (with provider/without provider) and validate that they returns the expected User object configuration.